quickjs.c

```diff
@@ -474,8 +472,8 @@ struct JSString {
     struct list_head link; /* string list */
 #endif
     union {
-        uint8_t str8[0]; /* 8 bit strings will get an extra null terminator */
-        uint16_t str16[0];
+        __extension__ uint8_t str8[0]; /* 8 bit strings will get an extra null terminator */
```

Comment:
Why does need the extension?

Reply:
Because zero length array members are a GNU extension, visibly supported by other compilers too, and we cannot use flexible arrays in `union` compound types.

Reply:
Ah I missed this was a flexible array too.